xapian-letor/Makefile.am

```diff
@@ -37,6 +37,30 @@ EXTRA_PROGRAMS =
 check-@PERCENT@:
 	cd tests && $(MAKE) $(AM_MAKEFLAGS) $@
 
+## --disable-shared as shared builds fail for me on x86-64 Debian etch.
```

Comment:
Debian _etch_? That was discontinued over 9 years ago! We should reassess this and update both here and in `-core` (or at least update the comment).

Reply:
And that comment originates in 06453ade9f9954fa826d4b8299b87d425fa24cbf over 10 years ago.

We should certainly reassess as a lot of time has passed, though I'd be OK to merge with the copied comment and then address this as a follow-on patch (the comment isn't actually wrong, just about really old versions).  Once these targets are available I can add letor coverage reports to lcov.xapian.org.

Reply:
I kicked off a xapian-core coverage build on Debian unstable with `--disable-shared` removed and it completed with a plausible coverage report generated.  It might be it fails on older platforms which are still in use though, so wider testing is probably needed.

Perhaps we should refactor to allow specifying arbitrary extra configure options, e.g.:

```
make coverage-reconfigure COVERAGE_CONFIGURE_ARGS=--disable-shared
```

Reply:
The VM which hosts lcov.xapian.org runs Debian stretch and that works without `--disable-shared` so I think removing it by default makes sense.  Generating coverage reports is really a developer thing, and developers should be running relatively up-to-date systems.  I'll push that change.

Reply:
Done, and also updated the developer guide.